collector/lib/NetworkSignalHandler.cpp

```diff
@@ -20,22 +20,32 @@ EventMap<Modifier> modifiers = {
         {"shutdown<", Modifier::REMOVE},
         {"connect<", Modifier::ADD},
         {"accept<", Modifier::ADD},
+        {"getsockopt<", Modifier::ADD},
     },
     Modifier::INVALID,
 };
 
 }  // namespace
 
 std::optional<Connection> NetworkSignalHandler::GetConnection(sinsp_evt* evt) {
+  auto* fd_info = evt->get_fd_info();
+  if (!fd_info) return std::nullopt;
+
+  if (fd_info->is_socket_failed()) {
+    return std::nullopt;
+  }
+
+  if (fd_info->is_socket_pending()) {
+    CLOG(DEBUG) << "Ignoring pending connection until we determine its status.";
+    return std::nullopt;
+  }
```

Comment:
so my understanding of this is that when the socket is pending, the result that we get on line 43 will be some errno value which is why we're first checking `fd_info`? Is this correct?

Regardless, I think it'd be helpful to have a little comment description of the socket lifecycle and expected results/behaviours

Reply:
> so my understanding of this is that when the socket is pending, the result that we get on line 43 will be some errno value which is why we're first checking `fd_info`? Is this correct?

This is correct, we now handle a case where a negative result has more meaning that just "fail". That is connect() returning with errno=E_INPROGRESS

> Regardless, I think it'd be helpful to have a little comment description of the socket lifecycle and expected results/behaviours

Indeed, this will be appreciated when I come back to this in 6 months :sweat_smile: 
I added some charts and comments, if you want to take a look -> https://github.com/stackrox/collector/commit/1a493e7a5214793b6e2e84c17bc2259720a8d347